kotlintest-tests/kotlintest-tests-core/src/test/kotlin/com/sksamuel/kotlintest/matchers/doubles/DoubleMatchersTest.kt

```diff
@@ -1,72 +1,1061 @@
-                    row(Double.MAX_VALUE),
-                    row(0.001),
-                    row(0.0),
-                    row(Double.NaN),
-                    row(Double.POSITIVE_INFINITY)
-            )
-
-            forAll(acceptedValues) {
-                it.shouldNotBeNegative()
-                it shouldNotBe negative()
+        
+      }
+      
+      "The non-numeric double" - {
+        "NaN" - {
+          
+          "Should not be exactly" - {
+            
+            "Any number" {
+              assertAll(numericDoubles) {
+                NaN shouldNotMatchExactly it
+              }
             }
-
-            val notAcceptedValues = table(
-                    headers("a"),
-                    row(-0.1),
-                    row(-1.0),
-                    row(Double.NEGATIVE_INFINITY)
-            )
-
-            forNone(notAcceptedValues) {
-                it.shouldNotBeNegative()
-                it shouldNotBe negative()
+            
+            "NaN" {
+              NaN shouldNotMatchExactly NaN
+            }
+            
+            "The infinities" {
+              NaN shouldNotMatchExactly POSITIVE_INFINITY
+              NaN shouldNotMatchExactly NEGATIVE_INFINITY
+            }
+          }
+        }
+        
+        "Positive Infinity" - {
+          
+          "Should be exactly" - {
+            "Itself" {
+              POSITIVE_INFINITY shouldExactlyMatch POSITIVE_INFINITY
+            }
+          }
+          
+          "Should not be exactly" - {
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                POSITIVE_INFINITY shouldNotMatchExactly it
+              }
+            }
+            
+            "Any other non-numeric double" {
+              POSITIVE_INFINITY shouldNotMatchExactly NEGATIVE_INFINITY
+              POSITIVE_INFINITY shouldNotMatchExactly NaN
+            }
+          }
+          
+        }
+        
+        "Negative Infinity" - {
+          
+          "Should be exactly" - {
+            "Itself" {
+              NEGATIVE_INFINITY shouldExactlyMatch NEGATIVE_INFINITY
+            }
+          }
+          
+          "Should not be exactly" - {
+            
+            "Any numeric double" {
+              assertAll(numericDoubles) {
+                NEGATIVE_INFINITY shouldNotMatchExactly it
+              }
+            }
+            
+            "Any other non-numeric double" {
+              NEGATIVE_INFINITY shouldNotMatchExactly POSITIVE_INFINITY
+              NEGATIVE_INFINITY shouldNotMatchExactly NaN
+            }
+          }
+        }
+      }
+      
+    }
+    
+    "Between matcher" - {
+      
+      "Every numeric double that is not Double.MAX_VALUE" - {
+        
+        "Should match between" - {
+          
+          "When it's equal to the first number of the range" - {
+            
+            "With tolerance" {
+              onDoubleRange { double, _ ->
```

Comment:
If you aren't using `valueToVariate`, then this is taking 10x the time without really helping. I would suggest useing `assertAll` directly in these cases.

Reply:
Good call